rasa/core/agent.py

```diff
@@ -707,6 +708,7 @@ async def load_data(
             exclusion_percentage=exclusion_percentage,
         )
 
+    @rasa.shared.utils.common.raise_on_unexpected_train
```

Comment:
what about doing a session wide fixture which monkeypatches this for tests? That way we don't have to have this in our production code.

Reply:
I was thinking maybe it would be useful in other scenarios, but you are right - best keep it out of production code.